donation_base/models/donation_tax_receipt.py

```diff
@@ -0,0 +1,108 @@
+# © 2014-2016 Barroux Abbey (http://www.barroux.org)
+# © 2014-2016 Akretion France (Alexis de Lattre <alexis.delattre@akretion.com>)
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from odoo import models, fields, api, _
+from odoo.exceptions import UserError
+
+
+class DonationTaxReceipt(models.Model):
+    _name = 'donation.tax.receipt'
+    _inherit = ['mail.thread']
+    _description = "Tax Receipt for Donations"
+    _order = 'id desc'
+    _rec_name = 'number'
+
+    number = fields.Char(string='Receipt Number')
+    date = fields.Date(
+        string='Date',
+        required=True,
+        default=fields.Date.context_today,
+        index=True
+    )
+    donation_date = fields.Date(
+        string='Donation Date'
+    )
+    amount = fields.Monetary(
+        string='Amount',
+        currency_field='currency_id'
+    )
+    currency_id = fields.Many2one(
+        'res.currency',
+        string='Currency',
+        required=True,
+        ondelete='restrict',
+        default=lambda self: self.env.user.company_id.currency_id.id
+    )
+    partner_id = fields.Many2one(
+        'res.partner',
+        string='Donor',
+        required=True,
+        ondelete='restrict',
+        domain=[('parent_id', '=', False)],
+        index=True
+    )
+    company_id = fields.Many2one(
+        'res.company',
+        string='Company',
+        required=True,
+        default=lambda self: self.env['res.company'].
+        _company_default_get('donation.tax.receipt')
+    )
+    print_date = fields.Date(string='Print Date')
+    type = fields.Selection([
+        ('each', 'One-Time Tax Receipt'),
+        ('annual', 'Annual Tax Receipt')],
+        string='Type',
+        required=True
+    )
+
+    # Maybe we can drop that code with the new seq management on v9
+    @api.model
+    def create(self, vals=None):
```

Comment:
Can you do it the way it is usually done now? With the default "/"

Reply:
not mandatory, but would be nice to have it identical in every module:
```
            if values.get('name', '/') == '/':
                seq = self.env['ir.sequence']
                values['name'] = seq.with_context(
                     date=date).next_by_code('donation.tax.receipt') or '/'
```

I would propose to do this after the migration merge, as all other modules for v12 depend on this one.
How is your opinion on this @jarroyomorales ?

Reply:
I think it can be done in the migration itself

Reply:
@nikul-serpentcs how would you like to proceed?

Reply:
> I think it can be done in the migration itself

Yes Sure

Reply:
@nikul-serpentcs will you do it soon? otherwise I would like to merge this PR so we can proceed with all depending PRs for 12.0 and create an improvement issue for this topic